diff --git a/src/main/kotlin/com/github/noamm9/features/impl/dungeon/ScoreCalculator.kt b/src/main/kotlin/com/github/noamm9/features/impl/dungeon/ScoreCalculator.kt
index 427bc5a..d51597c 100644
--- a/src/main/kotlin/com/github/noamm9/features/impl/dungeon/ScoreCalculator.kt
+++ b/src/main/kotlin/com/github/noamm9/features/impl/dungeon/ScoreCalculator.kt
@@ -19,6 +19,7 @@ object ScoreCalculator: Feature("Shows the score of the dungeon run.") {
     private val hudElement by ToggleSetting("HUD Element")
     val sendMimic by ToggleSetting("Send Mimic Message")
     val sendPrince by ToggleSetting("Send Prince Message")
+    val sendBat by ToggleSetting("Send Bat Message")
 
     private val sendMsg270 by ToggleSetting("270 score message").section("270")
     private val msg270 by TextInputSetting("Message", "270 Score!").showIf { sendMsg270.value }
diff --git a/src/main/kotlin/com/github/noamm9/features/impl/dungeon/map/MapRenderer.kt b/src/main/kotlin/com/github/noamm9/features/impl/dungeon/map/MapRenderer.kt
index 4f4d4c0..3f9aeba 100644
--- a/src/main/kotlin/com/github/noamm9/features/impl/dungeon/map/MapRenderer.kt
+++ b/src/main/kotlin/com/github/noamm9/features/impl/dungeon/map/MapRenderer.kt
@@ -66,11 +66,14 @@ object MapRenderer: HudElement() {
         val cryptsStr = colorCodeByPercent(ScoreCalculation.cryptsCount, 6) + "Crypts: ${ScoreCalculation.cryptsCount}"
         val scoreStr = "&eScore: ${colorizeScore(ScoreCalculation.score)}&r"
         val deathsStr = "&cDeaths: ${colorCodeByPercent(ScoreCalculation.deathCount, 4, true)}${ScoreCalculation.deathCount}&r"
-        val mimicStr = "&cM: ${if (ScoreCalculation.mimicKilled) "&a&l✔&r" else "&c&l✖&r"}"
-        val princeStr = "&eP: ${if (ScoreCalculation.princeKilled) "&a&l✔&r" else "&c&l✖&r"}"
+        val bonusStr = buildString {
+            append(if (ScoreCalculation.mimicKilled) "&aM &f| " else "&cM &f| ")
+            append(if (ScoreCalculation.princeKilled) "&aP &f| " else "&cP &f| ")
+            append(if (ScoreCalculation.batKilled) "&aB" else "&cB")
+        }
 
         val line1 = "$secretsStr    $cryptsStr"
-        val line2 = "$scoreStr   $deathsStr   $mimicStr $princeStr"
+        val line2 = "$scoreStr   $deathsStr   $bonusStr"
 
         ctx.pose().translate(width / 2f, 128f)
         Render2D.drawCenteredString(ctx, line1, 0f, - 4f, scale = 0.7f)
diff --git a/src/main/kotlin/com/github/noamm9/utils/dungeons/map/handlers/ScoreCalculation.kt b/src/main/kotlin/com/github/noamm9/utils/dungeons/map/handlers/ScoreCalculation.kt
index 0cba1a3..69cf61e 100644
--- a/src/main/kotlin/com/github/noamm9/utils/dungeons/map/handlers/ScoreCalculation.kt
+++ b/src/main/kotlin/com/github/noamm9/utils/dungeons/map/handlers/ScoreCalculation.kt
@@ -15,6 +15,7 @@ import com.github.noamm9.utils.dungeons.map.core.RoomState
 import com.github.noamm9.utils.equalsOneOf
 import com.github.noamm9.utils.location.LocationUtils
 import com.github.noamm9.websocket.WebSocket
+import com.github.noamm9.websocket.packets.S2CPacketDungeonBat
 import com.github.noamm9.websocket.packets.S2CPacketDungeonMimic
 import com.github.noamm9.websocket.packets.S2CPacketDungeonPrince
 import net.minecraft.network.protocol.Packet
@@ -52,6 +53,7 @@ object ScoreCalculation {
 
     var mimicKilled = false
     var princeKilled = false
+    var batKilled = false
     var deathCount = 0
     var foundSecrets = 0
     var cryptsCount = 0
@@ -75,6 +77,7 @@ object ScoreCalculation {
         var bScore = cryptsCount.coerceAtMost(5)
         if (mimicKilled && (LocationUtils.dungeonFloorNumber ?: 0) > 5) bScore += 2
         if (princeKilled) bScore += 1
+        if (batKilled) bScore += 1
         if (DungeonUtils.isPaul()) bScore += 10
         val bonusScore = bScore
 
@@ -174,6 +177,18 @@ object ScoreCalculation {
                 }
             }
 
+            if (! batKilled && batMessages.any { msg.contains(it) }) {
+                batKilled = true
+
+                if (DungeonListener.dungeonTeammatesNoSelf.isNotEmpty()) {
+                    WebSocket.send(S2CPacketDungeonBat)
+                }
+
+                if (ScoreCalculator.enabled && ScoreCalculator.sendBat.value) {
+                    ChatUtils.sendPartyMessage("Bat Killed")
+                }
+            }
+
             if ((LocationUtils.dungeonFloorNumber ?: 0) > 5 && ! LocationUtils.inBoss && mimicMessages.any { msg.contains(it) }) {
                 mimicKilled = true
             }
@@ -186,6 +201,7 @@ object ScoreCalculation {
         had270 = false
         mimicKilled = false
         princeKilled = false
+        batKilled = false
         deathCount = 0
         foundSecrets = 0
         cryptsCount = 0
@@ -240,4 +256,10 @@ object ScoreCalculation {
         "prince killed", "prince slain", "prince killed!",
         "a prince falls. +1 bonus score"
     )
+
+    private val batMessages = setOf(
+        "bat dead", "bat dead!", "\$skytils-dungeon-score-bat$",
+        "bat killed", "bat slain", "bat killed!",
+        "a bat has slain. +1 bonus score",
+    )
 }
\ No newline at end of file
diff --git a/src/main/kotlin/com/github/noamm9/websocket/PacketRegistry.kt b/src/main/kotlin/com/github/noamm9/websocket/PacketRegistry.kt
index a541ef2..a9c2e46 100644
--- a/src/main/kotlin/com/github/noamm9/websocket/PacketRegistry.kt
+++ b/src/main/kotlin/com/github/noamm9/websocket/PacketRegistry.kt
@@ -11,6 +11,7 @@ object PacketRegistry {
         register<S2CPacketDungeonDoor>("dungeondoor")
         register<S2CPacketDungeonMimic>("dungeonmimic")
         register<S2CPacketDungeonPrince>("dungeonprince")
+        register<S2CPacketDungeonBat>("dungeonbat")
         register<S2CPacketDungeonRoom>("dungeonroom")
         register<S2CPacketM7Dragon>("m7dragon")
         register<S2CPacketRoomSecrets>("dungeonroomsecrets")
